File: Participacipacion_4/PruebaMiEntero.java
package Participacipacion_4;
public class PruebaMiEntero {
    public static void main(String[] args) {
        MiEntero num = new MiEntero(7);

        System.out.println("Valor: " + num.getValor());
        System.out.println("¿Es par? " + num.esPar());
        System.out.println("¿Es impar? " + num.esImpar());
        System.out.println("¿Es primo? " + num.esPrimo());

        System.out.println("¿5 es par? " + MiEntero.esPar(5));
        System.out.println("¿5 es primo? " + MiEntero.esPrimo(5));

        MiEntero otro = new MiEntero(7);
        System.out.println("¿num = otro? " + num.equals(otro));

        char[] arr = {'1', '2', '3'};
        System.out.println("Parse char[]: " + MiEntero.parseInt(arr));
        System.out.println("Parse String: " + MiEntero.parseInt("456"));
    }
}

File: Participacipacion_4/PruebaMiEntero.py
from MiEntero import MiEntero

num = MiEntero(7)

print("Valor:", num.get_valor())
print("¿Es par?", num.es_par())
print("¿Es impar?", num.es_impar())
print("¿Es primo?", num.es_primo())

print("¿5 es par?", MiEntero.es_par_int(5))
print("¿5 es primo?", MiEntero.es_primo_int(5))

otro = MiEntero(7)
print("¿num = otro?", num.equals_obj(otro))

arr = ['1', '2', '3']
print("Parse char[]:", MiEntero.parse_char_array(arr))
print("Parse String:", MiEntero.parse_string("456"))